pkg/wave/decoder_test.go

```diff
@@ -0,0 +1,313 @@
+package wave
+
+import (
+	"encoding/binary"
+	"math"
+	"reflect"
+	"testing"
+)
+
+func TestCalculateChunkInfo(t *testing.T) {
+	testCases := map[string]struct {
+		chunk      []byte
+		channels   int
+		sampleSize int
+		expected   ChunkInfo
+		expectErr  bool
+	}{
+		"InvalidChunkSize1": {
+			chunk:      make([]byte, 3),
+			channels:   2,
+			sampleSize: 2,
+			expected:   ChunkInfo{},
+			expectErr:  true,
+		},
+		"InvalidChunkSize2": {
+			chunk:      make([]byte, 4),
+			channels:   2,
+			sampleSize: 4,
+			expected:   ChunkInfo{},
+			expectErr:  true,
+		},
+		"InvalidChannels": {
+			chunk:      nil,
+			channels:   0,
+			sampleSize: 2,
+			expected:   ChunkInfo{},
+			expectErr:  true,
+		},
+		"InvalidSampleSize": {
+			chunk:      nil,
+			channels:   2,
+			sampleSize: 0,
+			expected:   ChunkInfo{},
+			expectErr:  true,
+		},
+		"Valid1": {
+			chunk:      nil,
+			channels:   2,
+			sampleSize: 2,
+			expected: ChunkInfo{
+				Len:          0,
+				Channels:     2,
+				SamplingRate: 0,
+			},
+			expectErr: false,
+		},
+		"Valid2": {
+			chunk:      make([]byte, 8),
+			channels:   2,
+			sampleSize: 4,
+			expected: ChunkInfo{
+				Len:          1,
+				Channels:     2,
+				SamplingRate: 0,
+			},
+			expectErr: false,
+		},
+		"Valid3": {
+			chunk:      make([]byte, 4),
+			channels:   1,
+			sampleSize: 2,
+			expected: ChunkInfo{
+				Len:          2,
+				Channels:     1,
+				SamplingRate: 0,
+			},
+			expectErr: false,
+		},
+	}
+
+	for testCaseName, testCase := range testCases {
```

Comment:
```suggestion
	for testCaseName, testCase := range testCases {
		testCase := testCase
```
for parallel test safety.

Reply:
Ah, thanks. I forgot about that.